# serratia-code
The script enables systematic identification of Type VI Secretion System (T6SS) loci and their associated orthologous groups in *Serratia* genomes.

## Usage
Serratia_locus_finder.py detects T6SS loci from the output file generated by *Proteinortho v6.3.1*. You can modify the script as needed to adjust the definition of core T6SS components.

```shell
python3 serratia_locus_finder.py <filename.tsv>
```
**Example:** 
|       Species       |  Genes   |                   Alg.-Conn.              |   Query Gene ID   |   Target Gene ID   |
|:----------------------:| :---------: | :----------------------------------------------------------: |:--------------:|:--------------:|
|     2     |     4    |   0.5         | GCA_949794035.1_EBP3064_genome_genomic_04703,GCA_949794035.1_EBP3064_genome_genomic_00984 | VirG_234,VirG_235 |
|     2     |     3    |   0.33        | GCA_949794035.1_EBP3064_genome_genomic_03101 | TssA_244,TssA_345 |

After identifying the distribution of T6SS across *Serratia* genomes, you can calculate the presence/absence (P/A) values of all orthologous groups. First, cluster all *Serratia* proteins using *CD-HIT v4.8.1*, and save the output as a plain text file. Rename this file to input.txt, which will be used as the input for the cdhit-to-matrix.py script. This script will generate an output file named output_matrix.txt, also in TXT format, representing the presence/absence matrix of all orthologous clusters.

## Usage

```shell
python3 cdhit-to-matrix.py <input.txt> <output_matrix.txt>
```
**Example:** 
|       Clusters           |    Genes    |         
|:------------------------:| :---------: | 
|>Cluster 0 |
|0   |    9456aa, >GCA_016771795.1_ASM1677179v1_genomic_01131... *|
|1   |    9446aa, >GCA_030130165.1_ASM3013016v1_genomic_01136... at 98.96%|
|>Cluster 1 |
|0   |   6291aa, >GCA_000476315.1_UTAD54_1_genomic_04671... at 88.92%|
|1   |    7504aa, >GCA_001006005.1_ASM100600v1_genomic_02670... at 89.15%|

According to the script annotations, prepare the 'orthologue_file.txt', which serves as the input file for Comparative_T6SS.py. This script identifies T6SS<sup>serratia-1b</sup>-associated orthologous proteins.

```shell
python3 Comparative_T6SS.py <orthologue_file.txt> <presence_genomes.txt> <absence_genomes.txt>
```
**Example:** 
|Groups|   GCA_946228335.1 |GCA_030771795.1 |GCA_033365975.1 |GCA_016812305.1 |GCA_028616515.1| GCA_015999175.1| GCA_020887645|
|:------------------------:| :---------: | :--------------------------: |:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|
|Group1 | *     |  *  |    *   |    *  |     * |      *   |    *    |  
|Group2 | GCA_946228335.1_26968_7_47_genomic_01651     |   *     |  *  |     *  |     *    |   *    |  *    | 


'presence_genomes.txt' and 'absence_genomes.txt' are lists of genome names  for T6SS<sup>serratia-1b</sup>+/-, respectively. After identifying orthologous groups that are statistically associated with the presence of T6SS<sup>serratia-1b</sup>, the find_connected.py script can be used to detect all adjacent proteins in these clusters.

```shell
python3 find_connected.py <proteins_ID>
```
**Example:** 
|Protein ID|
|:------------------------:|
|GCA_000264275.1_ASM26427v1_genomic_00347|
|GCA_000442455.1_S.marcescens_LCT-SM166_genomic_04316|
|GCA_000513215.1_DB11_genomic_02635|
|GCA_000521905.1_Serr_marc_BIDMC_50_V1_genomic_03009|

